### PR TITLE
WIP: Change py-solc to py-solc-x

### LINKED ISSUE #142: Update contracts to 0.6.2 with eth-brownie or directly with py-solc-x
We've had the issue of managing 0.4.24+ versions with py-solc before. Seems like there's a convenient truffle-like project in development that can help us manage 0.5+ versions through py-solc-x.

Update all our contracts to latest 0.6.2 version.

Links:
https://eth-brownie.readthedocs.io/en/stable/
https://pypi.org/project/py-solc-x/

### PR DESCRIPTION
**Pull request by uivlis**
Fixes #142. I've managed to update the contracts to version `0.6.2` using `py-solc-x`, but I've stumbled upon a very strange error: during `truffle migrate`, I get this error for `Migrations.sol`:

```
1_initial_migration.js
======================

   Deploying 'Migrations'
   ----------------------

Error:  *** Deployment Failed ***

"Migrations" exceeded the block limit (with a gas value you set).
   * Block limit:  0x6691b7
   * Gas sent:     17592186044415
   * Try:
      + Sending less gas.
      + Setting a higher network block limit if you are on a
        private network or test client (like ganache).
```
Was wondering whether you have seen this before?

### CODE REVIEW

**Comment by posix4e** [comment]
@uivlis alas i see it all the time and it's not  the best error. I assume this ran against your local blockchain?

**Comment by uivlis** [comment]
Yes, during `bin/test`.

**Comment by posix4e** [comment]
You'll wanna make sure the ganache in docker-compose.yaml and the truffle version are up to date

**Comment by uivlis** [comment]
Hmm, I see the truffle version is the latest. I've updated ganache in the previous commit, so that would not be the cause of the issue.

File: hmt_escrow/eth_bridge.py [hunk 1]
```diff
@@ -28,9 +28,10 @@
 ])
 
 # See more details about the eth-kvstore here: https://github.com/hCaptcha/eth-kvstore
-KVSTORE_CONTRACT = Web3.toChecksumAddress(
-    os.getenv("KVSTORE_CONTRACT",
-              "0xbcF8274FAb0cbeD0099B2cAFe862035a6217Bf44"))
+# KVSTORE_CONTRACT = Web3.toChecksumAddress(
+#      os.getenv("KVSTORE_CONTRACT", "0xbcF8274FAb0cbeD0099B2cAFe862035a6217Bf44"))
+
+KVSTORE_CONTRACT = Web3.toChecksumAddress("0xbcF8274FAb0cbeD0099B2cAFe862035a6217Bf44")
```
Comment (rbval) [inline hmt_escrow/eth_bridge.py]: You'll need to read the `KVSTORE_CONTRACT` from the env, `0xbcF8274FAb0cbeD0099B2cAFe862035a6217Bf44` is a default that's not actually working.

File: truffle.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ module.exports = {
       host: ETH_HOST || '127.0.0.1',
       port: ETH_PORT || 9545,
       network_id: '*',
+      gas: 0xfffffffffff,
```
Comment (rbval) [inline truffle.js]: try removing these two

**Comment by uivlis** [comment]
Thanks @rbval for the suggestions, but the issue persisted...

**Comment by rbval** [comment]
Which version of truffle do you have? I ran your branch with `truffle@5.1.12`

**Comment by uivlis** [comment]
Yep, I ran it with that version, too.

**Comment by uivlis** [comment]
It's odd that's the `Migrations.sol` contract that throws the error... Do you happen to have other ideas what might be the cause of this issue?

**Comment by posix4e** [comment]
This needs a rebase

**Comment by posix4e** [comment]
I pulled your changes into https://github.com/hCaptcha/hmt-escrow/pull/147. Don't worry you still get credit. Let's see if we can push the new one over the finish line.

**Pull request closed without merging**